server/src/internalClusterTest/java/org/elasticsearch/reservedstate/service/SnaphotsAndFileSettingsIT.java

```diff
@@ -83,7 +89,20 @@ private void writeJSONFile(String node, String json) throws Exception {
         Path tempFilePath = createTempFile();
 
         Files.write(tempFilePath, Strings.format(json, version).getBytes(StandardCharsets.UTF_8));
-        Files.move(tempFilePath, fileSettingsService.operatorSettingsFile(), StandardCopyOption.ATOMIC_MOVE);
+        int retryCount = 0;
+        do {
+            try {
+                // this can fail on Windows because of timing
+                Files.move(tempFilePath, fileSettingsService.operatorSettingsFile(), StandardCopyOption.ATOMIC_MOVE);
+                return;
+            } catch (IOException e) {
+                if (retryCount == 4) { // retry 5 times
```

Comment:
Is it worth logging something indicating it's hit the retry?

Reply:
Ah yeah, good idea. I'll add that. Thanks!